Plugins/Public/event/Main.cpp

```diff
@@ -703,6 +774,23 @@ void __stdcall GFGoodBuy_AFTER(struct SGFGoodBuyInfo const &gbi, unsigned int iC
 					}
 				}
 
+				// Is the event limited by Ship class?
+				if (i->second.isClassRestricted)
+				{
+					//PrintUserCmdText(iClientID, L"Checking ship class requirements....");
+					Archetype::Ship* TheShipArch = Archetype::GetShip(Players[iClientID].iShipArchetype);
+					wstring& classname = shipclassnames.find(TheShipArch->iShipClass)->second;
```

Comment:
what if TheShipArch->iShipClass isn't found in shipclassnames? and shouldn't the comparison take place against ->first?

Reply:
Comparison happens against they key in find(), he's just inline extracting the value of ->second.

I agree that assuming a match will be found might cause issues in the future.

Do just the find, ensure it's not end(), and proceed appropriately.